dashboard/v1.1/src/pages/Config/Config.tsx

```diff
@@ -0,0 +1,231 @@
+    switch (intervalName) {
+      case 'ping':
+        setToggleResults({ ...toggleResults, ping: !toggleResults['ping'] });
+        break;
+      case 'jitter':
+        setToggleResults({
+          ...toggleResults,
+          jitter: !toggleResults['jitter']
+        });
+        break;
+      case 'req-res-delay-and-monitoring':
+        setToggleResults({
+          ...toggleResults,
+          'req-res-delay-and-monitoring': !toggleResults[
+            'req-res-delay-and-monitoring'
+          ]
+        });
+        break;
+      default:
+        break;
+    }
+  };
+
+  return (
+    <div>
+      <Grid container spacing={4}>
+        {configIntervals?.map(interval => {
+          const { test, duration, unit } = interval;
+          return (
+            <Grid item lg={3} sm={6} xl={3} xs={12} key={test}>
+              <Card className={classes.cardStyle}>
+                <CardContent>
+                  <Grid container style={{ justifyContent: 'space-between' }}>
+                    <Grid item>
+                      <Typography gutterBottom variant="h6">
+                        {truncate(
+                          test.charAt(0).toUpperCase() + test.slice(1),
+                          14
+                        )}
+                      </Typography>
+                    </Grid>
+                    <Grid item>
+                      {toggleResults[test] ? (
+                        <Tooltip title="Cancel" style={{ cursor: 'pointer' }}>
+                          <CloseIcon onClick={e => handleToggle(e, test)} />
+                        </Tooltip>
+                      ) : (
+                        <Tooltip title="Edit" style={{ cursor: 'pointer' }}>
+                          <EditIcon onClick={e => handleToggle(e, test)} />
+                        </Tooltip>
+                      )}
+                    </Grid>
+                  </Grid>
+                  <IntervalDetails
+                    reFetch={fetchConfigIntervals}
+                    toggleResults={toggleResults}
+                    durationValue={duration}
+                    intervalName={test}
+                  />
+                  <Typography variant="body1">{unit}</Typography>
+                </CardContent>
+              </Card>
+            </Grid>
+          );
+        })}
+      </Grid>
+      <SearchTable
+        title=""
+        columns={columns}
+        data={getTableRoutes(configRoutes)}
+        editable={{
+          onRowUpdate: (newData, oldData) =>
+            new Promise((resolve, reject) => {
+              setTimeout(() => {
+                // const index = oldData.tableData.id;
+
+                resolve();
+              }, 1000);
+            }),
+          onRowDelete: oldData =>
+            new Promise((resolve, reject) => {
+              setTimeout(() => {
+                // const index = oldData.tableData.id;
+
+                resolve();
+              }, 1000);
```

Comment:
can you pls tell why are the resolves inside the timeouts?

Reply:
This will be changed to a normal API request because as you said it's hard to edit/delete routes

Reply:
oh, sure